### PR TITLE
Add multicontainer and regular pnnl dopf

### PR DESCRIPTION
**Pull request by josephmckinsey**
~Currently the multicontainer download fails~ Fixed.

### CODE REVIEW

**Comment by Tylores** [comment]
- cvxpy is not included in requirements.txt so oedisi build fails for the regular scenario.
- the scenario file includes an "existing_feeder_file" which doesn't have a fallback by the feeder federate if it doesn't exist. 
- docker compose up doesn't work from the project directory and needs to be called from build to work. It should default to build and only need specification if you build to a different folder. 
- There is a PORT environment variable that causes the oedisi broker container to fail.
oedisi-multi-error

**Comment by josephmckinsey** [comment]
> * cvxpy is not included in requirements.txt so oedisi build fails for the regular scenario.

Fixed. Plus I switched to a nested requirements.txt

>     * the scenario file includes an "existing_feeder_file" which doesn't have a fallback by the feeder federate if it doesn't exist.

I changed this to downloading instead. I do want to warn that the PV power in the gadal_ieee123 right now is still messed up.

>     * docker compose up doesn't work from the project directory and needs to be called from build to work. It should default to build and only need specification if you build to a different folder.

I don't really have a solution to this since we sort of need the yaml file to be with the build folders

>     * There is a PORT environment variable that causes the oedisi broker container to fail.
> 
> 
> oedisi-multi-error

I think this is part of Aadil's updates on his image. Using --pull=never should probably work, and I think --build on docker compose up should work too.

**Comment by josephmckinsey** [comment]
@AadilLatif I just copied all that code from https://github.com/pnnl/oedisi_dopf, so I'm not sure about fixing it up...

**Comment by josephmckinsey** [comment]
To myself: add a CI test.

File: lindistflow_federate/lindistflow.py [hunk 1]
```diff
@@ -0,0 +1,944 @@
...
+            counteq += 1
+            # Phase C
+            pcc, qcc = -2 * z[2, 2][0], -2 * z[2, 2][1]
+            pca, qca = (
+                -(-z[0, 2][0] + math.sqrt(3) * z[0, 2][1]),
+                -(-z[0, 2][1] - math.sqrt(3) * z[0, 2][0]),
+            )
+            pcb, qcb = (
+                -(-z[1, 2][0] - math.sqrt(3) * z[1, 2][1]),
+                -(-z[0, 2][1] + math.sqrt(3) * z[1, 2][0]),
```
Comment (AadilLatif) [inline lindistflow_federate/lindistflow.py:442]: index is incorrect

**Review by AadilLatif — APPROVED** [review]

**Pull request merged**